bitcoin/src/blockdata/locktime/absolute.rs

```diff
@@ -460,6 +553,176 @@ impl FromHexStr for Height {
     }
 }
 
+impl ops::Add<relative::Height> for Height {
+    type Output = Height;
+    #[inline]
+    fn add(self, rhs: relative::Height) -> Self::Output {
+        match self.checked_add(rhs.value().into()) {
+            Some(height) => height,
+            None => panic!("attempted to add height with overflow"),
+        }
+    }
+}
+
+impl ops::Add<u32> for Height {
+    type Output = Height;
+    #[inline]
+    fn add(self, rhs: u32) -> Self::Output {
+        let height = self.0 + rhs;
+        Height::from_consensus(height).expect("attempted to add with overflow")
+    }
+}
+
+impl ops::Add<Height> for u32 {
+    type Output = Height;
+    #[inline]
+    fn add(self, rhs: Height) -> Self::Output { rhs + self }
+}
+
+impl ops::Sub<relative::Height> for Height {
+    type Output = Height;
+    #[inline]
+    fn sub(self, rhs: relative::Height) -> Self::Output {
+        match self.checked_sub(rhs.value().into()) {
+            Some(height) => height,
+            None => panic!("attempted to sub height with overflow"),
+        }
+    }
+}
+
+impl ops::Sub<u32> for Height {
+    type Output = Height;
+    #[inline]
+    fn sub(self, rhs: u32) -> Self::Output {
+        let height = self.0 - rhs;
+        Height::from_consensus(height).expect("non-panicing sub as a valid height")
+    }
+}
+
+impl ops::Sub<Height> for u32 {
+    type Output = Height;
+    #[inline]
+    fn sub(self, rhs: Height) -> Self::Output { rhs - self }
+}
+
+impl ops::Mul<relative::Height> for Height {
```

Comment:
This is definitely wrong. What does multiplying an instant with duration even mean? Or blocks squared? Even if we wanted it it'd need `HeightSquared` output type.